packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -165,6 +173,13 @@ export default function TypographyPanel( {
 	panelId,
 	defaultControls = DEFAULT_CONTROLS,
 } ) {
+	// If paneId has a value, it means this panel will be rendered inside
+	// the block sidebar. The text alignment UI for individual blocks is rendered
+	// in the block toolbar, so disable support if it's not a global style.
+	if ( panelId ) {
+		settings.typography.textAlign = false;
+	}
```

Comment:
This was the point that bothered me the most. This prevents duplicate UI in inserted blocks, but there may be a better approach.

![duplicate-text-align-ui](https://github.com/WordPress/gutenberg/assets/54422211/036fe213-6f35-4608-9a84-80da0184c48e)